src/App.tsx

```diff
@@ -1,61 +1,132 @@
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUserById = (userId: number): User | null => (
+  usersFromServer.find((user) => (user.id === userId)) || null
+);
+
+const getUserByName = (userName: string): User | null => (
+  usersFromServer.find((user) => (user.name === userName)) || null
+);
+
+const newTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(newTodos);
+  const [selectedUser, setSelectedUser] = useState('');
+  const [isTitleValid, setIsTitleValid] = useState(true);
+  const [isUserValid, setIsUserValid] = useState(true);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
+    setIsTitleValid(Boolean(value));
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setSelectedUser(value);
+    setIsUserValid(Boolean(value));
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setIsTitleValid(Boolean(title.trim()));
+    setIsUserValid(Boolean(selectedUser));
+
+    if (!title.trim() || !selectedUser) {
+      return;
+    }
+
+    setTodos((currentTodos) => {
+      const maxToDoId = Math.max(...currentTodos.map(todo => todo.id));
+      const newUser = getUserByName(selectedUser);
+
+      return [
+        ...currentTodos,
+        {
+          id: maxToDoId + 1,
+          title,
+          completed: false,
+          userId: newUser ? newUser.id : null,
+          user: newUser,
+        },
+      ];
```

Comment:
Let's create a variable fro new todo to improve readability

```suggestion
      cosnt newTodo = {...}
      return [ ...currentTodos, newTodo];
```